### DISCUSSION TITLE
Lifecycle callbacks are not called

### DISCUSSION THREAD

**Discussion by user576g** [sceneview/sceneview#181] [Q&A]
I wonna AR view for Flutter
When I add module to the app, there is no exception, but just **black** view on the **screen**.

Put your attention that:
1) calbacks like ARSceneView#onArSessionCreated or SceneView#onCreate are **not** called;
2) it's flutter plugin, so FlutterFragmentActivity is used instead of AppCompatActivity.
FlutterFragmentActivity extends FragmentActivity.

I'll be glad to read any suggestions.

Code:

package com.epam.arquest.ar_quest_plugin

import android.content.Context
import android.graphics.Color
import android.view.View
import android.widget.TextView
import io.flutter.plugin.platform.PlatformView
import io.github.sceneview.ar.ArSceneView
import io.github.sceneview.ar.arcore.ArSession
import io.github.sceneview.math.Position
import io.github.sceneview.math.Rotation
import io.github.sceneview.math.Scale
import io.github.sceneview.node.ModelNode

internal class ARView(
        private val context: Context,
        id: Int,
        creationParams: Map<String?, Any?>?
) : PlatformView {

    private val textView: TextView

    init {
        textView = TextView(context)
        textView.textSize = 72f
        textView.setBackgroundColor(Color.rgb(255, 255, 255))
        textView.text = "Error! context is NOT ContextWrapper!"
    }

    override fun getView(): View {
        val arSceneView = ArSceneView(context)
        arSceneView.addChild(
                ModelNode(
                        position = Position(x = 0.0f, y = 0.0f, z = -4.0f),
                        rotation = Rotation(y = 90.0f),
                        scale = Scale(0.5f)
                )
        )
        print("asdf")
        arSceneView.onArSessionCreated = { arSession: ArSession ->
            // this is not called
            print("asdf")
            println(arSession)
        }
        arSceneView.onArSessionFailed = { exception: Exception ->
            // this is not called
            print("asdf")
            println(exception.stackTrace)
        }

        return arSceneView
    }

    override fun dispose() {}
}

![telegram-cloud-photo-size-2-5431576390718832304-y](https://user-images.githubusercontent.com/52647767/200830611-5c97a362-b046-451e-9507-13ed486e82a8.jpg)

Upd: If someone has the same issue. I solved it in this way: I replaced dependency 'io.github.sceneview:arsceneview:0.9.0' with "com.gorisse.thomas.sceneform:sceneform:1.21.0". I handled activity lifecycle callbacks by myself and called arSceneView.resume() in onResume() method.
So, it seems like io.github.sceneview:arsceneview can be used in application module only.

**Comment by Kliuchko** [comment]
I faced the same issue with AR view for Flutter. First of all I created my own lifecycle owner and then called it when needed. Everything worked, but it was not comfortable. But now that the guys have updated the library 'io.github.sceneview:arsceneview:2.0.2' - everything works without any custom lifecycle classes.

**Comment by sameerjj** [comment]
have u checked out https://github.com/SceneView/sceneview-flutter

**Comment by ThomasGorisse** [comment]
In v4.0.0-rc.1, SceneView is fully integrated with Compose lifecycle. Instead of manual lifecycle callbacks, use Compose effects:

```kotlin
SceneView {
    DisposableEffect(Unit) {
        // Called when the scene enters composition
        onDispose {
            // Called when the scene leaves composition
        }
    }
    LaunchedEffect(Unit) {
        // Coroutine scoped to the SceneView lifecycle
    }
}
```

The Compose approach replaces the old Fragment/View lifecycle callbacks (onResume, onPause, etc.) with standard Compose side effects, which are more reliable and easier to reason about.
